### PR TITLE
cilium: Datapath transparent encryption CI test

### PR DESCRIPTION
**Pull request by jrfastab**
Add initial baseline tests to DatapathConfiguration so that all CI
runs now do some enabled-ipsec tests.

Signed-off-by: John Fastabend <john.fastabend@gmail.com>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/7303)

### CODE REVIEW

**Comment by jrfastab** [comment]
Requires https://github.com/cilium/cilium/pull/7302 to pass.

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
Something failed here all the Datapath failed and were not able to connect to cilium.sock and etcd connections failed. I'm going to retry tests while I look into it.:
 https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/10593/

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
This failure on 4.9 (testing net-next now) is a result of a e1000 hang. Locally I fixed this by upgrading to 6.0
```
2019-03-08T16:47:12,290474+0000 e1000 0000:00:03.0 enp0s3: Detected Tx Unit Hang
                                  Tx Queue             <0>
                                  TDH                  <d2>
                                  TDT                  <d4>
                                  next_to_use          <d4>
                                  next_to_clean        <d2>
                                buffer_info[next_to_clean]
                                  time_stamp           <100044b0b>
                                  next_to_watch        <d3>
                                  jiffies              <100044db0>
                                  next_to_watch.status <0>
2019-03-08T16:47:14,306524+0000 e1000 0000:00:03.0 enp0s3: Detected Tx Unit Hang
                                  Tx Queue             <0>
                                  TDH                  <d2>
                                  TDT                  <d4>
                                  next_to_use          <d4>
                                  next_to_clean        <d2>
                                buffer_info[next_to_clean]
                                  time_stamp           <100044b0b>
                                  next_to_watch        <d3>
                                  jiffies              <100044fa8>
                                  next_to_watch.status <0>
2019-03-08T16:47:16,323312+0000 e1000 0000:00:03.0 enp0s3: Detected Tx Unit Hang
                                  Tx Queue             <0>
                                  TDH                  <d2>
                                  TDT                  <d4>
                                  next_to_use          <d4>
                                  next_to_clean        <d2>
                                buffer_info[next_to_clean]
                                  time_stamp           <100044b0b>
                                  next_to_watch        <d3>
                                  jiffies              <1000451a0>
                                  next_to_watch.status <0>
2019-03-08T16:47:17,986142+0000 ------------[ cut here ]----------
```
Going to have to wait for vbox upgrade to do 4.9 testing it appears.

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
FQDN failure unrelated and already tracked so retry.

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
Star wars demo failed... https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/10734/testReport/junit/k8s-1/8/K8sDemosTest_Tests_Star_Wars_Demo/

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
testing 4.9 kernel now

**Comment by jrfastab** [comment]
please-test-me

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

**Comment by aanm** [comment]
test-flannel

File: test/k8sT/DatapathConfiguration.go [hunk 1]
```diff
@@ -29,10 +29,12 @@ var _ = Describe("K8sDatapathConfig", func() {
 
 	var kubectl *helpers.Kubectl
 	var demoDSPath string
+	var ipsecDSPath string
 
 	BeforeAll(func() {
 		kubectl = helpers.CreateKubectl(helpers.K8s1VMName(), logger)
 		demoDSPath = helpers.ManifestGet("demo_ds.yaml")
+		ipsecDSPath = helpers.ManifestGet("ipsec_ds.yaml")
```
Comment (eloycoto) [inline test/k8sT/DatapathConfiguration.go]: could you change this to ipsec_keys.yaml instead ipsecDsPath?

Something like: 
```
ipsecKeysPath = helpers.ManifestGet("ipsec_keys.yaml") 
```

File: test/k8sT/DatapathConfiguration.go [hunk 2]
```diff
@@ -104,6 +108,23 @@ var _ = Describe("K8sDatapathConfig", func() {
 			Expect(status.IntOutput()).Should(Equal(3), "Did not find expected number of entries in BPF tunnel map")
 		}
 
+		It("Check connectivity with transparent encryption and VXLAN encapsulation", func() {
+			switch helpers.GetCurrentIntegration() {
```
Comment (eloycoto) [inline test/k8sT/DatapathConfiguration.go]: you can change these lines to: 
```
			SkipIfFlannel()
```

File: test/k8sT/DatapathConfiguration.go [hunk 3]
```diff
@@ -104,6 +108,23 @@ var _ = Describe("K8sDatapathConfig", func() {
...
+			case helpers.CIIntegrationFlannel:
+				Skip(fmt.Sprintf(
+					"Cilium in %q mode is not supported with transparent encryption and VxLAN. Skipping test.",
+					helpers.CIIntegrationFlannel))
+				return
+			}
+
+			deployCilium("cilium-ds-patch-vxlan-ipsec.yaml")
+			kubectl.Exec("ip x p f")
```
Comment (eloycoto) [inline test/k8sT/DatapathConfiguration.go]: I'm not sure why this commands are here, if you can remove or add a comment will be great ;-)

File: test/k8sT/DatapathConfiguration.go [hunk 4]
```diff
@@ -151,6 +172,14 @@ var _ = Describe("K8sDatapathConfig", func() {
 		})
 	})
 
+	Context("Transparent encryption with IPv4Only", func() {
+		It("Check connectivity with transparent encryption enabled and IPv6 disabled", func() {
+			deployCilium("cilium-ds-patch-ipv4-only-ipsec.yaml")
```
Comment (eloycoto) [inline test/k8sT/DatapathConfiguration.go]: shouldn't this be skipped in flannel case?

**Review by eloycoto — CHANGES_REQUESTED** [review]

**Comment by jrfastab** [comment]
test-me-please

File: test/k8sT/DatapathConfiguration.go (hunk 4 above)
Comment (jrfastab) [reply test/k8sT/DatapathConfiguration.go]: Skipped for now will debug why this is not working and add it back later.

File: test/k8sT/DatapathConfiguration.go (hunk 3 above)
Comment (jrfastab) [reply test/k8sT/DatapathConfiguration.go]: Those were there for debugging reasons originally. They are not needed, will remove

File: test/k8sT/DatapathConfiguration.go (hunk 2 above)
Comment (jrfastab) [reply test/k8sT/DatapathConfiguration.go]: thanks will do.

File: test/k8sT/DatapathConfiguration.go (hunk 1 above)
Comment (jrfastab) [reply test/k8sT/DatapathConfiguration.go]: Sounds good copied above.

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
A couple errors on IPSec tests rerun to see if we get the same ones

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
```
2019-03-21T16:24:43.945874465Z 2019-03-21 16:24:43.945751 I | embed: rejected connection from "127.0.0.1:53512" (error "tls: failed to verify client's certificate: x509: certificate specifies an incompatible key usage", ServerName "")
```

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
no longer needed, added via other PRs.

**Pull request closed without merging**